client/components/image-preloader/index.jsx

```diff
@@ -72,9 +74,18 @@ module.exports = React.createClass( {
 	onLoadComplete: function( event ) {
 		this.destroyLoader();
 
-		this.setState( {
-			status: 'load' === event.type ? LoadStatus.LOADED : LoadStatus.FAILED
-		} );
+		if ( event.type !== 'load' ) {
+			this.setState( { status: LoadStatus.FAILED } );
```

Comment:
Should we trigger the callbacks after the state change completes (to avoid weird races)?

```
this.setState( { status: LoadStatus.FAILED }, () => {
	if ( this.props.onError ) {
		this.props.onError( event );
	}
} );
```